### PR TITLE
Add lower dimensional elements along the faces contained in a side set

### LINKED ISSUE #9371: MeshModifier to add face elements along side sets
### Description of the enhancement or error report
Add lower dimensional elements along the faces of a chosen sideset.

### Rationale for the enhancement or information for reproducing the error
Easy setup of meshes to couple mixed dimensional problems.

### Identified impact
Added capability.

### PR DESCRIPTION
**Pull request by dschwen**
Closes #9371

### CODE REVIEW

File: framework/src/meshmodifiers/MeshSideSet.C [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  // Reference the the libMesh::MeshBase
+  auto & mesh = _mesh_ptr->getMesh();
+  auto & boundary_info = mesh.get_boundary_info();
+
+  // get IDs of all boundaries to be broken
+  std::set<BoundaryID> mesh_boundary_ids;
+  if (isParamValid("boundaries"))
+  {
+    auto & boundary_names = getParam<std::vector<BoundaryName>>("boundaries");
+    for (auto i = beginIndex(boundary_names); i < boundary_names.size(); ++i)
```
Comment (permcody) [inline framework/src/meshmodifiers/MeshSideSet.C]: Could use a range-based loop here.

File: framework/src/meshmodifiers/MeshSideSet.C [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  }
+  else
+    mesh_boundary_ids = boundary_info.get_boundary_ids();
+
+  _mesh_ptr->buildBndElemList();
+  for (auto it = _mesh_ptr->bndElemsBegin(); it != _mesh_ptr->bndElemsEnd(); ++it)
+    if (mesh_boundary_ids.count((*it)->_bnd_id) > 0)
+    {
+      Elem * elem = (*it)->_elem;
+      unsigned short int s = (*it)->_side;
```
Comment (permcody) [inline framework/src/meshmodifiers/MeshSideSet.C]: Well that's a mouthful - could just use auto here?

File: framework/include/meshmodifiers/MeshSideSet.h [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+class MeshSideSet : public MeshModifier
+{
+public:
+  MeshSideSet(const InputParameters & parameters);
+
+  virtual void modify() override;
+
+protected:
+  /// Block ID to assign to the region
+  SubdomainID _block_id;
```
Comment (permcody) [inline framework/include/meshmodifiers/MeshSideSet.h]: Should be `const`

File: framework/src/meshmodifiers/MeshSideSet.C [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+      // build element from the side
+      std::unique_ptr<Elem> side(elem->build_side(s, false));
+      side->processor_id() = elem->processor_id();
```
Comment (permcody) [inline framework/src/meshmodifiers/MeshSideSet.C]: Interesting, is this necessary?

**Review by permcody — DISMISSED** [review]
Looks fine, just a couple of minor comments you might want to fix.

File: framework/src/meshmodifiers/MeshSideSet.C (hunk 4 above)
Comment (dschwen) [inline framework/src/meshmodifiers/MeshSideSet.C]: Looks like it:

```
      // build element from the side
      std::unique_ptr<Elem> side(elem->build_side(s, false));
      if (side->processor_id() != elem->processor_id())
        mooseError("yes, it is necessary, Cody!");
      side->processor_id() = elem->processor_id();
```

```
[11][schwd][~/Programs/moose/test]> ./run_tests -p 4 --re=mesh_side_set

Oversized Jobs:

mesh_modifiers/mesh_side_set.test............................................................ FAILED (ERRMSG)
mesh_modifiers/mesh_side_set.test: Working Directory: /Users/schwd/Programs/moose/test/tests/mesh_modifiers/mesh_side_set
mesh_modifiers/mesh_side_set.test: Running command: mpiexec -n 4 /Users/schwd/Programs/moose/test/moose_test-opt --n-threads=1 -i test.i --error --no-gdb-backtrace
mesh_modifiers/mesh_side_set.test:
mesh_modifiers/mesh_side_set.test:
mesh_modifiers/mesh_side_set.test:
mesh_modifiers/mesh_side_set.test:
mesh_modifiers/mesh_side_set.test:
mesh_modifiers/mesh_side_set.test: *** ERROR ***
mesh_modifiers/mesh_side_set.test: yes, it is necessary, Cody!
```

**Comment by dschwen** [comment]
Ok, I addressed the comments and applied the suggestions to the various files I looked at to write this. This should keep things consistent.

**Review by permcody — DISMISSED** [review]

**Comment by dschwen** [comment]
Let me add a doc page real quick...

File: docs/app_syntax.yml [hunk 5]
```diff
@@ -53,7 +53,7 @@ locations:
             - /Materials
             - /Mesh/Partitioner
             - /Mesh/MortarInterfaces
-            - /MeshModifiers
+            # - /MeshModifiers
```
Comment (dschwen) [inline docs/app_syntax.yml]: Fuuuuuuuuuuuuuuuuuuuuuuuuu! And that's why I opened #9383 @aeslaughter ...

**Comment by aeslaughter** [comment]
My recommendation continues to be, if you a

**Review by permcody — APPROVED** [review]

**Pull request merged**